### PR TITLE
[ADD] inventory: add reservation methods docs

### PR DESCRIPTION
**Pull request by brse-odoo**
This PR is an [ADD] to add `reservation_methods` docs into Odoo's Inventory documentation, nested under the `shipping_receiving` section, specific to 17.0.

Here's what I pushed up initially:
- New doc for overview of reservations methods
- Updated toc-tree and added new image folder for first doc

**_Important:_**
- _Please ignore the errors on rst. line 98, as the referenced docs have not been merged yet, but will as a part of this PR._
- _Please ignore any merge conflicts, as those are expected and will be resolved as I continue to push up to this PR._

### Edit: Updated Merge Order
- First, merge this doc. Then, merge:
- [PR #7876](https://github.com/odoo/documentation/pull/7876). Then, merge:
- [PR #7886](https://github.com/odoo/documentation/pull/7886). Then, merge:
- [PR #7925](https://github.com/odoo/documentation/pull/7925) (last in the merge order)

### CODE REVIEW

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:28]:
```suggestion
      quantity **must** be reserved manually.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:32]:
```suggestion
      A specific number of days can be selected; this is the maximum number of days **before**
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (hojo-odoo) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:92]:
```suggestion
Operations Types`, and then clicking on a specific operations type.
```

**Review by hojo-odoo — APPROVED** [review]
Hey @brse-odoo,

Another great doc! Just a few minor suggestions for you to take a look at, and then you can move it to the next round of review.

Thanks,
hojo

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:28]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:32]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:92]: Suggestion implemented ✅

**Review by ksc-odoo — APPROVED** [review]
@brse-odoo great work on this doc. I just finished my Final Review and have 'approved' this PR, provided that you take a look at my feedback, and make the necessary adjustments, before moving it to Tech Review. You'll notice that some of my suggestions are purely optional. If you have any questions, don't hesitate to reach out. Thanks! 👍

**Comment by StraubCreative** [comment]
Hi @brse-odoo would you like to address KC's feedback items or move forward to tech review and merge stages? In either case please make your revisions (if any) and tag the next person in the review chain 🙂 
As well, the failing CI check will need to be resolved at some point...
Thanks!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:9]: Brevity's always good with me! I'll add this change in.

Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:43]: That makes sense to me. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:66]: Yeah, that actually does make more sense in this instance. I'll go ahead and make the change.

Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:78]: I agree, it does work better. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:81]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:85]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:83]: Nice catch. Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:87]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:39]: That's a good call to add that in there. I added an additional `.. note::` admonition before the last image to explain this:

.. note::
   When the :guilabel:`Type of Operation` is changed to *Receipt* on an :guilabel:`Operations Type`
   form, reservation methods are **not** available.

Suggestion implemented ✅

**Comment by brse-odoo** [comment]
> @brse-odoo great work on this doc. I just finished my Final Review and have 'approved' this PR, provided that you take a look at my feedback, and make the necessary adjustments, before moving it to Tech Review. You'll notice that some of my suggestions are purely optional. If you have any questions, don't hesitate to reach out. Thanks! 👍

**Comment by brse-odoo** [comment]
Thank you @ksc-odoo for the quick review!

**Comment by brse-odoo** [comment]
Before sending to technical review, going to resolve the merge conflicts by adding custom anchors (while implementing review changes) for the docs following this one in the merge order: [PR #7876](https://github.com/odoo/documentation/pull/7876), [PR #7886](https://github.com/odoo/documentation/pull/7886), and [PR #7925](https://github.com/odoo/documentation/pull/7925).

Will update this PR once again to resolve merge conflicts, then send to tech review 👍

**Comment by StraubCreative** [comment]
> Before sending to technical review, going to resolve the merge conflicts by adding custom anchors (while implementing review changes) for the docs following this one in the merge order: [PR #7876](https://github.com/odoo/documentation/pull/7876), [PR #7886](https://github.com/odoo/documentation/pull/7886), and [PR #7925](https://github.com/odoo/documentation/pull/7925).
> 
> Will update this PR once again to resolve merge conflicts, then send to tech review 👍

Hi @brse-odoo the quick fix to this is to remove any/all links to docs that don't exist yet and then add them back in, one by one, on each proceeding PR after this doc is published.

What that means here is: remove the **toctree** block at line 105-110, along with any other hyperlink references to non-existent docs in the RST. Push back up and then the checks should pass after that.

**Comment by brse-odoo** [comment]
@StraubCreative Thank you for the advice to fix the merge conflicts. 👍 Alright @samueljlieber, this PR is ready for your technical review! Thank you.

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:73]: Suggesting to use the **Apps and modules Install** documentation instead of re-explaining how to install the modules :)
```suggestion
The two required applications that **must** be :ref:`installed <general/install>` to use reservation
methods are the *Sales* and *Inventory* apps.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:99]: Please use :guilabel:s for drop-down menu options in an active instruction :)
```suggestion
   When the :guilabel:`Type of Operation` is changed to :guilabel:`Receipt` on an
   :guilabel:`Operations Type` form, reservation methods are **not** available.
```

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:90]: Wouldnt the _Manufacturing_ and _Repairs_ apps be **Required applications** in this case? If so, it would make sense to me to include these apps in the previous instructions to install the _Sales_ and _Inventory_ apps. 

Doing so would also simplify the bullet points to show just the config of **For resupply subcontractor** and **For internal transfers** instead of both installing apps and config.

I also may not be understanding this section correctly, so please let me know your thoughts. There isnt anything technically wrong here, just voicing my thoughts 🙂

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (samueljlieber) [inline content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:3]: No need to include these since there are no sub-docs _yet_ 🙂
```suggestion
```

**Review by samueljlieber — CHANGES_REQUESTED** [review]
Hi @brse-odoo! Nice job on this brand new Inventory doc, the PR is looking good! I have a few corrections and some points for discussion, please let me know your thoughts! 

Once these have been addressed, tag me again for a quick look. Thank you!

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:73]: That's a nice way to save some time! Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:99]: Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:3]: Nice catch! Suggestion implemented ✅

File: content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst
Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:90]: My logic with this section was that you don't actually need to use reservation methods for these functions if you don't want, since it's most commonly used on delivery orders. So the info in this list is just telling people how to go about setting up the "optional" extras, so to speak, if that makes sense.

Comment (brse-odoo) [reply content/applications/inventory_and_mrp/inventory/shipping_receiving/reservation_methods.rst:90]: I added some words for clarification to try and make this a bit more clear

**Review by StraubCreative — APPROVED** [review]
Looks good @brse-odoo
Once this + the FWPs are all merged in, let's continue building out this sub-directory next week with #7876 and beyond 🚀 
.....
@robodoo r+

**Pull request closed without merging**
